src/test/java/com/github/wovnio/wovnjava/HeadersTest.java

```diff
@@ -248,6 +248,27 @@ public void testRedirectLocationPathNestedDirectory() {
         assertEquals("https://example.com/ja/dir1/dir2/", h.redirectLocation("ja"));
     }
 
+    public void testRedirectLocationWithSitePrefixPathBasePathOnly() {
+        Headers h = makeHeaderWithSitePrefixPath("/global", "/global/");
+        assertEquals("https://example.com/global", h.redirectLocation("en")); // `en` is defaultLang
+        assertEquals("https://example.com/global/ja/", h.redirectLocation("ja"));
+        assertEquals("https://example.com/global/garbage/", h.redirectLocation("garbage"));
+    }
+
+    public void testRedirectLocationWithSitePrefixPathMatchingPath() {
+        Headers h = makeHeaderWithSitePrefixPath("/global/tokyo/", "/global/");
+        assertEquals("https://example.com/global/tokyo/", h.redirectLocation("en")); // `en` is defaultLang
+        assertEquals("https://example.com/global/ja/tokyo/", h.redirectLocation("ja"));
+        assertEquals("https://example.com/global/garbage/tokyo/", h.redirectLocation("garbage"));
+    }
+
+    public void testRedirectLocationWithSitePrefixPathNonmatchingPath() {
```

Comment:
Here's how I would name these tests that test a single scenario:
`redirectLocation_WithSitePrefixPathBasePathOnly_AddsLanguageToUrl`
`redirectLocation_WithSitePrefixPathMatchingPath_AddsLanguageToUrl`
`redirectLocation_WithSitePrefixPathNonmatchingPath_DoesNotModifyUrl`

Just some food for thought! I think it helps readability
https://medium.com/@lorenzen.jacob/unit-test-naming-conventions-875786f53790
https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html